### PR TITLE
separate sagemaker jobs

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/training/buildspec.yml [hunk 1]
```diff
@@ -28,19 +28,19 @@ context:
       target: deep_learning_container.py
 
 images:
```
Comment (tejaschumbalkar) [inline pytorch/training/buildspec.yml:30]: to be reverted before PR merge

File: src/start_testbuilds.py [hunk 2]
```diff
@@ -99,38 +90,48 @@ def run_test_job(commit, codebuild_project, images_str=""):
     LOGGER.debug(f"env_overrides dict: {env_overrides}")
 
     client = boto3.client("codebuild")
-    return client.start_build(
-        projectName=codebuild_project,
-        environmentVariablesOverride=env_overrides,
-        sourceVersion=commit,
-    )
+    LOGGER.debug(f"Starting build job {codebuild_project}")
```
Comment (tejaschumbalkar) [inline src/start_testbuilds.py:93]: To be reverted once the cb jobs are available

File: test/dlc_tests/conftest.py [hunk 3]
```diff
@@ -7,8 +7,6 @@
 import time
 import uuid
 import boto3
```
Comment (tejaschumbalkar) [inline test/dlc_tests/conftest.py:9]: remove unused imports

File: test/testrunner.py [hunk 4]
```diff
@@ -1,11 +1,9 @@
 import json
 import os
```
Comment (tejaschumbalkar) [inline test/testrunner.py:2]: remove unused imports

File: dlc_developer_config.toml [hunk 5]
```diff
@@ -24,33 +24,32 @@ huggingface_trcomp_mode = false
 # Do remember to revert it back to False before merging any PR (including TRCOMP dedicated PR)
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:24]: to be reverted before PR merge

File: dlc_developer_config.toml [hunk 6]
```diff
@@ -62,12 +61,14 @@ ec2_tests_on_heavy_instances = false
 ### Off by default
 sagemaker_local_tests = false
 
-# SM remote test valid values:
-# "off" --> do not trigger sagemaker remote tests (default)
-# "standard" --> run standard sagemaker remote tests from test/sagemaker_tests
-# "rc" --> run release_candidate_integration tests
-# "efa" --> run efa sagemaker tests
-sagemaker_remote_tests = "off"
+# run standard sagemaker remote tests from test/sagemaker_tests
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:64]: updated this section

File: dlc_developer_config.toml [hunk 7]
```diff
@@ -24,33 +24,32 @@ huggingface_trcomp_mode = false
...
   safety_check_test = false
   ecr_scan_allowlist_feature = false
-ecs_tests = true
-eks_tests = true
-ec2_tests = true
+ecs_tests = false
+eks_tests = false
+ec2_tests = false
+# Please only set it to True if you are preparing a Benchmark related PR
+ec2_benchmark_tests = false
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:52]: Added new config

File: pytorch/training/buildspec.yml [hunk 8]
```diff
@@ -84,21 +84,21 @@ images:
   #   target: sagemaker
```
Comment (tejaschumbalkar) [inline pytorch/training/buildspec.yml:84]: to be reverted before PR merge

File: test/test_utils/sagemaker.py [hunk 9]
```diff
@@ -238,7 +238,7 @@ def generate_sagemaker_pytest_cmd(image, sagemaker_test_type):
         f"--processor {processor} {docker_base_arg} {sm_remote_docker_base_name} --tag {tag} "
```
Comment (tejaschumbalkar) [inline test/test_utils/sagemaker.py:238]: to be reverted before PR merge

File: test/testrunner.py [hunk 10]
```diff
@@ -402,7 +392,7 @@ def main():
                 raise Exception(f"EKS cluster {eks_cluster_name} is not in active state")
 
         # Execute dlc_tests pytest command
-        pytest_cmd = ["-s", "-rA", test_path, f"--junitxml={report}", "-n=auto"]
+        pytest_cmd = ["-s", "-rA", test_path, f"--junitxml={report}", "-n=auto", "--collect-only"]
```
Comment (tejaschumbalkar) [inline test/testrunner.py:395]: reverted before PR merge

**Review by saimidu — APPROVED** [review]
LGTM. Please revert temporary changes whenever ready to merge.

File: test/dlc_tests/sanity/quick_checks/test_dlc_developer_config.py [hunk 11]
```diff
@@ -50,14 +53,18 @@ def test_developer_config_wrappers_defaults():
...
     assert config.is_sm_remote_test_enabled() is False
+    assert config.is_sm_efa_test_enabled is False
+    assert config.is_sm_rc_test_enabled is False
     assert config.get_sagemaker_remote_efa_instance_type() == ""
     assert config.is_ecs_test_enabled() is True
     assert config.is_eks_test_enabled() is True
     assert config.is_ec2_test_enabled() is True
+    assert config.is_ec2_benchmark_test_enabled is False
+    assert config.is_sm_remote_tests_enabled
+    assert config.is_sm_benchmark_test_enabled is False
```
Comment (saimidu) [inline test/dlc_tests/sanity/quick_checks/test_dlc_developer_config.py:67]: Please change all of these to function calls.

**Review by saimidu — COMMENTED** [review]
LGTM except for the comment reg. the config tests.

File: test/dlc_tests/sanity/quick_checks/test_dlc_developer_config.py (hunk 11 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/quick_checks/test_dlc_developer_config.py]: good catch, addressed

**Review by saimidu — APPROVED** [review]

**Review by saimidu — APPROVED** [review]

**Pull request merged**